test/rules/avoid_init_to_null.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(AvoidInitToNullTest);
+  });
+}
+
+@reflectiveTest
+class AvoidInitToNullTest extends LintRuleTest {
+  @override
+  String get lintRule => 'avoid_init_to_null';
+
+  test_lintOnNullable() async {
```

Comment:
You can name tests however you like, but the style we've developed in the analyzer seems to work fairly well. For example, we try to keep the names as short as possible. To that end I'd drop the "lintOn" and "NoLintOn" prefixes (and always use lowercase after an underscore). More generally, the test name should describe the condition being tested, not the outcome of the test. The names of the "assert" methods tend to be enough for that in practice.

I'd also add "_topLevelVariable" in keeping with the pattern used in the rest of the methods.